hnn_core/cell_response.py

```diff
@@ -476,3 +505,37 @@ def read_spikes(fname, gid_ranges=None):
         cell_response.update_types(gid_ranges)
 
     return cell_response
+
+
+def read_spikes_hdf5(fname, gid_ranges=None):
+    """Read spiking activity from a hdf5 file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the hdf5 file (e.g., '<pathname>/spk.hdf5') to be read.
+    gid_ranges : dict of lists or range objects | None
+        Dictionary with keys 'evprox1', 'evdist1' etc.
+        containing the range of Cell or input IDs of different
+        cell or input types. If None, data object should contain
+        spike type.
+
+    Returns
+    ----------
+    cell_response : CellResponse
+        An instance of the CellResponse object.
+    """
+
+    # A check to ensure file exists
+    if not os.path.exists(fname):
+        raise NameError("File not found at path %s" % (fname,))
+    data = read_hdf5(fname)
+    # Write checks to ensure data object contain all required
+    # attributes
+    print(data)
+    cell_response = CellResponse(spike_times=data['spike_times'],
+                                 spike_gids=data['spike_gids'],
+                                 spike_types=data['spike_types'])
+    if gid_ranges is not None:
```

Comment:
I think this applies only to the old function ... in the new function, hopefully everything is already saved in the file, so user does not have to provide the `gid_ranges`

Reply:
Yes everything is saved in the hdf5 file so will remove `gid_ranges`